remoteconfig.go

```diff
@@ -0,0 +1,199 @@
+	}
+
+	var remoteConfigArray []remoteConfig
+	err = json.NewDecoder(resp.Body).Decode(&remoteConfigArray)
+	if err != nil {
+		return err
+	}
+
+	remoteConfigCache, err := createRemoteConfigCache(remoteConfigArray)
+	if err != nil {
+		return err
+	}
+
+	sg.remoteConfigCache = remoteConfigCache
+	return nil
+}
+
+func (sg *Service) shouldIgnoreRequestRemoteConfig(req *http.Request) bool {
+	domain := domainutils.Domain(req.Host)
+	if domain == "" {
+		domain = domainutils.Subdomain(req.Host)
+	}
+	if domain == "" {
+		return false
+	}
+
+	endpoints := sg.remoteConfigCache[domain]
+	if endpoints == nil {
+		return false
+	}
+	for _, endpoint := range endpoints {
+		testVal, err := marshalEndpointLocationValue(req, endpoint.Location)
+		if err != nil {
+			sg.options.OnError(err)
+			continue
+		}
+		testByteArray := []byte(fmt.Sprintf("%v", testVal))
+		match := endpoint.Regex.Match(testByteArray)
+		if match {
+			// Any value in remote config cache should not be ingested
+			return true
+		}
+	}
+	return false
+}
+
+func createRemoteConfigCache(remoteConfigArray []remoteConfig) (map[string][]endpointCacheVal, error) {
+	remoteConfigMap := map[string][]endpointCacheVal{}
+	for _, config := range remoteConfigArray {
+		cacheVal := []endpointCacheVal{}
+		for _, endpoint := range config.Endpoints {
+			// NOTE: the remote config today is only used to block requests
+			// from supergood ingest. We only need to track those endpoints that
+			// should be ignored. All others that arent in remoteConfigMap can be allowed
+			if endpoint.EndpointConfiguration.Action != "Ignore" {
+				continue
+			}
+			if endpoint.MatchingRegex.Regex == "" || endpoint.MatchingRegex.Location == "" {
+				continue
+			}
+			regex, err := regexp.Compile(endpoint.MatchingRegex.Regex)
+			if err != nil {
+				return nil, err
+			}
+			endpointCacheVal := endpointCacheVal{
+				Regex:    regex,
+				Location: endpoint.MatchingRegex.Location,
+				Action:   endpoint.EndpointConfiguration.Action,
+			}
+			cacheVal = append(cacheVal, endpointCacheVal)
+		}
+		remoteConfigMap[config.Domain] = cacheVal
+	}
+	return remoteConfigMap, nil
+}
+
+// mashalEndpointLocation value takes an endpoint location, which is used to uniquely classify
+// a request and stringifies the request object at that location
+func marshalEndpointLocationValue(req *http.Request, location string) (string, error) {
+	switch location {
+	case "subdomain":
+		return req.URL.Host, nil
+	case "url":
+		return req.URL.String(), nil
+	case "path":
+		return req.URL.Path, nil
+	case "domain":
+		return req.URL.Host, nil
+	case "request_body":
+		body, err := io.ReadAll(req.Body)
+		if err != nil {
+			return "", err
+		}
+		req.Body = io.NopCloser(bytes.NewBuffer(body))
+		return string(body), nil
+	case "request_headers":
+		return fmt.Sprint(req.Header), nil
+	default:
+		return "", errors.New("invalid location parameter for RegExp matching")
+	}
+}
```

Comment:
This is a weak implementation of matching regex's against go http request struct fields.